File: LinkedList/Introduction/LinkedList.cpp
#include "Node.h"

int main()
{

    // Initilizing nodes
    Node *head = new Node(1);
    Node *second = new Node(2);
    Node *third = new Node(3);

    // Linking Node
    head->next = second;  // 1 -> 2
    second->next = third; // 1 -> 2 -> 3

    Node *temp = head;

    while (temp != NULL)
    {
        cout << temp->data;

        temp = temp->next;

        if (temp != NULL)
            cout << " -> ";
    }
    cout << "\n";

    return 0;
}

File: LinkedList/CircularLinkedList/Traversal/LinkedList.java
package LinkedList.CircularLinkedList.Traversal;

public class LinkedList {

    // static Node head;

    static class Node {
        int data;
        Node next;

        Node(int data) {
            this.data = data;
        }
    }

    static void printList(Node head) {

        if (head == null) {
            System.out.println("Circular Linkedlist is Empty");
            return;
        }

        Node temp = head;

        do {

            System.out.print(temp.data);
            temp = temp.next;
            if (temp != head) {
                System.out.print(" -> ");
            }

        } while (temp != head);
        System.out.println();
    }

    public static void main(String[] args) {

        Node head = new Node(1);
        Node second = new Node(2);
        Node third = new Node(3);
        Node forth = new Node(4);
        Node fifth = new Node(5);

        // Linked one node to the next node
        head.next = second;
        second.next = third;
        third.next = forth;
        forth.next = fifth;
        fifth.next = head;

        /* To print the elements of the LinkedList */
        LinkedList.printList(head);
    }

}